nexosim/src/simulation/sim_init.rs

```diff
@@ -217,27 +220,129 @@ impl SimInit {
+    where
+        T: Message + DeserializeOwned + Clone + Send + 'static,
+        R: Message + Serialize + Send + 'static,
+    {
+        self.endpoint_registry.add_query_source(source, name)
+    }
+
+    /// Adds a query source to the endpoint registry without requiring `Message`
+    /// implementations for its query and response types.
+    ///
+    /// If the specified name is already in use for another query source, the
+    /// source provided as argument is returned in the error.
+    pub fn add_query_source_raw_endpoint<T, R>(
+        &mut self,
+        source: QuerySource<T, R>,
+        name: impl Into<String>,
+    ) -> Result<(), QuerySource<T, R>>
+    where
+        T: DeserializeOwned + Clone + Send + 'static,
+        R: Serialize + Send + 'static,
+    {
+        self.endpoint_registry.add_query_source_raw(source, name)
+    }
+
+    /// Adds an event sink to the endpoint registry.
+    ///
+    /// If the specified name is already in use for another event sink, the
+    /// event sink provided as argument is returned in the error.
+    pub fn add_event_sink_endpoint<S>(&mut self, sink: S, name: impl Into<String>) -> Result<(), S>
+    where
+        S: EventSinkReader + Send + Sync + 'static,
+        S::Item: Message + Serialize,
+    {
+        self.endpoint_registry.add_event_sink(sink, name)
+    }
+
+    /// Adds an event sink to the endpoint registry without requiring a
+    /// `Message` implementation for its item type.
+    ///
+    /// If the specified name is already in use for another event sink, the
+    /// event sink provided as argument is returned in the error.
+    pub fn add_event_sink_raw_endpoint<S>(
+        &mut self,
+        sink: S,
+        name: impl Into<String>,
+    ) -> Result<(), S>
+    where
+        S: EventSinkReader + Send + Sync + 'static,
+        S::Item: Serialize,
+    {
+        self.endpoint_registry.add_event_sink_raw(sink, name)
+    }
+
+    fn build(self) -> (Simulation, EndpointRegistry) {
+        let (mut simulation, mut endpoint_registry) = (
+            Simulation::new(
+                self.executor,
+                self.scheduler_queue,
+                self.scheduler_registry,
+                self.time,
+                self.clock,
+                self.clock_tolerance,
+                self.timeout,
+                self.observers,
+                self.registered_models,
+                self.is_halted,
+            ),
+            self.endpoint_registry,
+        );
+        endpoint_registry.register_scheduler(&mut simulation.scheduler_registry);
+
+        (simulation, endpoint_registry)
     }
 
     /// Builds a simulation initialized at the specified simulation time,
     /// executing the [`Model::init`] method on all
     /// model initializers.
     ///
     /// The simulation object is returned upon success.
-    pub fn init(mut self, start_time: MonotonicTime) -> Result<Simulation, SimulationError> {
+    pub fn init(
```

Comment:
Is it a problem to name this `init_with_registry` and keep the old signature alongside?